### PR TITLE
feature(typography): add component

### PR DESCRIPTION
**Pull request by hvolschenk**

### CODE REVIEW

**Comment by hvolschenk** [comment]
Hiya @moog16

All part of the fun. Still made time to play some SC, hehe.

I hope it is fine that I add my components here, I just much rather prefer a single source of truth rather than us all maintaining separate libraries.

File: examples/roses/Feedback.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ import TopAppBar from '@material/react-top-app-bar';
 import TextField, {Input, HelperText} from '@material/react-text-field';
 import Button from '@material/react-button';
 import MaterialIcon from '@material/react-material-icon';
+import {Body2, Headline6} from '@material/react-typography';
```
Comment (moog16) [inline examples/roses/Feedback.js]: thanks for updating the example!

File: examples/roses/package.json [hunk 2]
```diff
@@ -40,6 +40,7 @@
     "@material/react-switch": "^0.5.0",
     "@material/react-text-field": "^0.5.1",
     "@material/react-top-app-bar": "^0.5.0",
+    "@material/react-typography": "^0.5.0",
```
Comment (moog16) [inline examples/roses/package.json]: change this to 0.0.0

File: packages/typography/Body1.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+const Body1 = (props) => {
+  const {children, className, tag: Tag, ...otherProps} = props;
+  const classes = classnames('mdc-typography', 'mdc-typography--body1', className);
+  return (
+    <Tag className={classes} {...otherProps}>{children}</Tag>
+  );
+};
+
+Body1.propTypes = {
+  children: PropTypes.node.isRequired,
```
Comment (moog16) [inline packages/typography/Body1.js]: I think children should be
```
PropTypes.oneOfType([string, element])
```

Thoughts?

File: packages/typography/README.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+  Overline,
+  Subtitle1,
+  Subtitle2,
+} from '@material/react-typography';
+
+class MyApp extends React.Component {
+  render() {
+    return (
+      <React.Fragment>
+        <Headline1>Kate Lockwell</Headline1>
```
Comment (moog16) [inline packages/typography/README.md]: NICE (we need a SC emoji)

File: packages/typography/Button.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+Button.propTypes = {
+  children: PropTypes.node.isRequired,
+  className: PropTypes.string,
+  tag: PropTypes.string,
+};
+
+Button.defaultProps = {
+  className: '',
+  tag: 'span',
```
Comment (moog16) [inline packages/typography/Button.js]: should this be a button?

File: packages/typography/package.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    "material design",
+    "material typography",
+    "materialtypography"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/material-components/material-components-web-react.git"
+  },
+  "dependencies": {
+    "@material/typography": "^0.40.1",
```
Comment (moog16) [inline packages/typography/package.json]: lets aim to put this into https://github.com/material-components/material-components-web-react/tree/rc7.0. Please update to 0.41.0

**Comment by hvolschenk** [comment]
@moog16 I have created the HOC, much better than a bunch of components that all look exactly the same. The requested changes have also been implemented. I have also added responses to some of your questions/concerns. Thanks for the feedback.

File: package.json [hunk 7]
```diff
@@ -46,7 +46,8 @@
           "tab-scroller",
           "infrastructure",
           "tab-indicator",
-          "text-field"
```
Comment (moog16) [inline package.json]: also add mdc-typography as a dep.

Comment (hvolschenk) [reply package.json]: I have added `typography` (on line _50_) already.

I also see that `"@material/typography": "^0.40.1"` was already on line _80_ of this file.

Anything I am missing?

Comment (moog16) [reply package.json]: AH! I meant `"@material/typography": "^0.40.1"`...thanks

File: packages/typography/Typography.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+import classnames from 'classnames';
```
Comment (moog16) [inline packages/typography/Typography.js]: rename this file to `typography.js` - lowercase

File: packages/typography/.npmignore [hunk 9]
```diff
@@ -0,0 +1,2 @@
+index.js
+Typography.js
```
Comment (moog16) [inline packages/typography/.npmignore]: rename this file to `typography.js` - lowercase

**Comment by moog16** [comment]
Also sign the PR too please

**Comment by hvolschenk** [comment]
I signed it

**Comment by moog16** [comment]
tests passing https://github.com/material-components/material-components-web-react/pull/407

**Pull request merged**